### PR TITLE
cmake: Silence policy CMP0023 warning.

### PR DESCRIPTION
**Pull request by undisputed-seraphim**

### CODE REVIEW

File: plugins/oderave/CMakeLists.txt [hunk 1]
```diff
@@ -50,17 +50,14 @@ if(ODE_LIBRARY_FOUND)
...
+  target_link_libraries(oderave PUBLIC libopenrave ${ODE_LIBRARY} PRIVATE boost_assertion_failed)
   set_target_properties(oderave PROPERTIES COMPILE_FLAGS "${PLUGIN_COMPILE_FLAGS} ${ODE_CXXFLAGS}" LINK_FLAGS "${PLUGIN_LINK_FLAGS}")
   install(TARGETS oderave DESTINATION ${OPENRAVE_PLUGINS_INSTALL_DIR} COMPONENT ${COMPONENT_PREFIX}plugin-oderave)
   
   if( MSVC )
     # have to install the DLLs
     install(DIRECTORY "${ODE_LINK_DIRS}/" DESTINATION bin COMPONENT ${COMPONENT_PREFIX}plugin-oderave FILES_MATCHING PATTERN "*-${MSVC_PREFIX}-*.dll")
   endif()
-else()
-  message(STATUS "Could not find ODE. Please install ODE (http://www.ode.org)")
```
Comment (cielavenir) [inline plugins/oderave/CMakeLists.txt:63]: is removing this ok?

File: plugins/qtcoinrave/CMakeLists.txt [hunk 2]
```diff
@@ -1,17 +1,18 @@
 ############################
 # qtcoinrave openrave plugin
 ############################
-find_package(Coin)
-find_package(SoQt)
+find_package(Coin QUIET)
+find_package(SoQt QUIET)
```
Comment (cielavenir) [inline plugins/qtcoinrave/CMakeLists.txt:5]: thank you for fixing my qtcoin patch properly :grin:

**Comment by cielavenir** [comment]
could you paste the autotester pipeline number? (yes only number not entire uri)

the thing is, due to -Wl,--as-needed being default in Debian 11 (and Ubuntu 20.04), changing the link order can have unexpected result.

File: plugins/oderave/CMakeLists.txt (hunk 1 above)
Comment (undisputed-seraphim) [reply plugins/oderave/CMakeLists.txt:63]: That message is redundant, the same message is printed if FindODE.cmake fails to find ODE.
Message is printed twice so I removed one.

**Comment by undisputed-seraphim** [comment]
254482

File: plugins/textserver/CMakeLists.txt [hunk 3]
```diff
@@ -3,12 +3,7 @@
 ###########################################
 add_library(textserver SHARED textserver.cpp textserver.h plugindefs.h)
 
-if( MSVC )
-  target_link_libraries(textserver libopenrave imm32 winmm ws2_32)
-else()
-  target_link_libraries(textserver libopenrave)
-endif()
-target_link_libraries(textserver PRIVATE boost_assertion_failed)
+target_link_libraries(textserver PUBLIC libopenrave $<$<CXX_COMPILER_ID:MSVC>:imm32> $<$<CXX_COMPILER_ID:MSVC>:winmm> $<$<CXX_COMPILER_ID:MSVC>:ws2_32> PRIVATE boost_assertion_failed)
```
Comment (rdiankov) [inline plugins/textserver/CMakeLists.txt:6]: @undisputed-seraphim congrats on making the one-liner harder to read, please revert back unless you have a good reason for doing it this way.

File: plugins/rmanipulation/CMakeLists.txt [hunk 4]
```diff
@@ -7,13 +7,10 @@ add_library(rmanipulation SHARED rmanipulation.cpp basemanipulation.cpp    plugi
...
   message(STATUS "boost regex found")
   set(PLUGIN_COMPILE_FLAGS "${PLUGIN_COMPILE_FLAGS} -DHAVE_BOOST_REGEX")
-  target_link_libraries(rmanipulation libopenrave ${Boost_REGEX_LIBRARY})
 else()
   message(STATUS "failed to find boost regex, please install it")
-  target_link_libraries(rmanipulation libopenrave)
 endif()
-target_link_libraries(rmanipulation PRIVATE boost_assertion_failed)
-
+target_link_libraries(rmanipulation PRIVATE boost_assertion_failed PUBLIC libopenrave $<$<BOOL:Boost_REGEX_FOUND>:${Boost_REGEX_LIBRARY}>)
```
Comment (rdiankov) [inline plugins/rmanipulation/CMakeLists.txt:13]: @undisputed-seraphim what's the point of making this harder to read?

File: plugins/rplanners/piecewisepolynomials/CMakeLists.txt [hunk 5]
```diff
@@ -3,8 +3,7 @@ message(STATUS "Eigen3 version: ${Eigen3_VERSION}")
 include_directories(${EIGEN3_INCLUDE_DIRS})
 
 add_library(piecewisepolynomials SHARED polynomialcommon.h polynomialtrajectory.h polynomialtrajectory.cpp polynomialchecker.h polynomialchecker.cpp interpolatorbase.h cubicinterpolator.h cubicinterpolator.cpp quinticinterpolator.h quinticinterpolator.cpp feasibilitychecker.h generalrecursiveinterpolator.h generalrecursiveinterpolator.cpp)
-target_link_libraries(piecewisepolynomials libopenrave)
-target_link_libraries(piecewisepolynomials PRIVATE boost_assertion_failed)
+target_link_libraries(piecewisepolynomials PRIVATE boost_assertion_failed PUBLIC libopenrave)
```
Comment (rdiankov) [inline plugins/rplanners/piecewisepolynomials/CMakeLists.txt:6]: @undisputed-seraphim in some cases you have PUBLIC libraries declared first, and in others you have PRIVATE first. please be consistent.

File: plugins/rmanipulation/CMakeLists.txt (hunk 4 above)
Comment (undisputed-seraphim) [reply plugins/rmanipulation/CMakeLists.txt:13]: It's... really not harder to read? Well, I guess I'll revert it if it's too hard.

File: plugins/rplanners/piecewisepolynomials/CMakeLists.txt (hunk 5 above)
Comment (undisputed-seraphim) [reply plugins/rplanners/piecewisepolynomials/CMakeLists.txt:6]: Noted.

Comment (undisputed-seraphim) [reply plugins/rplanners/piecewisepolynomials/CMakeLists.txt:6]: Done.

File: plugins/textserver/CMakeLists.txt (hunk 3 above)
Comment (undisputed-seraphim) [reply plugins/textserver/CMakeLists.txt]: I agree, and reverted.

**Comment by undisputed-seraphim** [comment]
254518

**Comment by rdiankov** [comment]
looks better~ thanks~

File: plugins/textserver/CMakeLists.txt [hunk 6]
```diff
@@ -3,12 +3,11 @@
 ###########################################
 add_library(textserver SHARED textserver.cpp textserver.h plugindefs.h)
 
-if( MSVC )
-  target_link_libraries(textserver libopenrave imm32 winmm ws2_32)
+if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+  target_link_libraries(textserver PRIVATE boost_assertion_failed PUBLIC libopenrave imm32 winmm ws2_32)
```
Comment (cielavenir) [inline plugins/textserver/CMakeLists.txt:7]: suggestion during working on another pull request https://github.com/rdiankov/openrave/pull/1115#issuecomment-1133591443

File: plugins/ikfastsolvers/CMakeLists.txt [hunk 7]
```diff
@@ -12,12 +12,11 @@ endif()
...
 add_library(ikfastsolvers SHARED ikfastsolvers.cpp ikfastmodule.cpp ikfastsolver.cpp plugindefs.h ${CMAKE_CURRENT_SOURCE_DIR}/../../python/ikfast.h)# ${ik_files})
-if( Boost_IOSTREAMS_FOUND )
-  target_link_libraries(ikfastsolvers libopenrave ${Boost_IOSTREAMS_LIBRARY} ${LAPACK_LIBRARIES})
+if (Boost_IOSTREAMS_FOUND)
+  target_link_libraries(ikfastsolvers PRIVATE boost_assertion_failed PUBLIC libopenrave ${LAPACK_LIBRARIES} ${Boost_IOSTREAMS_LIBRARY})
 else()
-  target_link_libraries(ikfastsolvers libopenrave ${LAPACK_LIBRARIES})
+  target_link_libraries(ikfastsolvers PRIVATE boost_assertion_failed PUBLIC libopenrave ${LAPACK_LIBRARIES})
 endif()
-target_link_libraries(ikfastsolvers PRIVATE boost_assertion_failed)
```
Comment (cielavenir) [inline plugins/ikfastsolvers/CMakeLists.txt:20]: @undisputed-seraphim 

```
if (Boost_IOSTREAMS_FOUND)
  target_link_libraries(ikfastsolvers PUBLIC libopenrave ${LAPACK_LIBRARIES} ${Boost_IOSTREAMS_LIBRARY})
else()
  target_link_libraries(ikfastsolvers PUBLIC libopenrave ${LAPACK_LIBRARIES})
endif()
target_link_libraries(ikfastsolvers PRIVATE boost_assertion_failed)
```

File: plugins/rmanipulation/CMakeLists.txt [hunk 8]
```diff
@@ -7,13 +7,11 @@ add_library(rmanipulation SHARED rmanipulation.cpp basemanipulation.cpp    plugi
...
   message(STATUS "boost regex found")
   set(PLUGIN_COMPILE_FLAGS "${PLUGIN_COMPILE_FLAGS} -DHAVE_BOOST_REGEX")
-  target_link_libraries(rmanipulation libopenrave ${Boost_REGEX_LIBRARY})
+  target_link_libraries(rmanipulation PRIVATE boost_assertion_failed PUBLIC libopenrave ${Boost_REGEX_LIBRARY})
 else()
   message(STATUS "failed to find boost regex, please install it")
-  target_link_libraries(rmanipulation libopenrave)
+  target_link_libraries(rmanipulation PRIVATE boost_assertion_failed PUBLIC libopenrave)
 endif()
-target_link_libraries(rmanipulation PRIVATE boost_assertion_failed)
```
Comment (cielavenir) [inline plugins/rmanipulation/CMakeLists.txt:15]: @undisputed-seraphim 

```
if( Boost_REGEX_FOUND )
  message(STATUS "boost regex found")
  set(PLUGIN_COMPILE_FLAGS "${PLUGIN_COMPILE_FLAGS} -DHAVE_BOOST_REGEX")
  target_link_libraries(rmanipulation PUBLIC libopenrave ${Boost_REGEX_LIBRARY})
else()
  message(STATUS "failed to find boost regex, please install it")
  target_link_libraries(rmanipulation PUBLIC libopenrave)
endif()
target_link_libraries(rmanipulation PRIVATE boost_assertion_failed)
```

**Pull request merged**